### ISSUE TITLE
"Device is not setup as a development device. Use Xcode to setup."

### ISSUE THREAD

**Issue by rgustavsson** [ios-driver/ios-driver#131]
Hey!

I admit that I'm a newbie to MAC and Xcode and can't seem to find an answer elsewhere.

I'm trying to run tests on a simulator and this is what I do:
1. Followed the setup instructions @ http://ios-driver.github.io/ios-driver/setup.html
2. Server is running:

java -jar ios-server-0.6.0.jar -host 10.99.255.141 -port 8084
53:27:287 INFO ApplicationStore.<init>  App archive folder:/Users/richard/Downloads/applications
53:28:747 INFO IOSServer.init  
Beta features enabled ( enabled by -beta flag ): false
Inspector: http://0.0.0.0:8084/inspector/
tests can access the server at http://0.0.0.0:8084/wd/hub
server status: http://0.0.0.0:8084/wd/hub/status
Connected devices: http://0.0.0.0:8084/wd/hub/devices/all
Applications: http://0.0.0.0:8084/wd/hub/applications/all
Capabilities: http://0.0.0.0:8084/wd/hub/capabilities/all
Monitoring '/Users/richard/Downloads/applications' for new applications
using xcode install : /Applications/Xcode.app
using IOS version 6.1
Archived apps /Users/richard/Downloads/applications
ios >= 6.0. Safari and hybrid apps are supported.
## Applications :

```
CFBundleName=Safari,CFBundleVersion=8536.25
```

2013-08-29 16:53:28.748:INFO::jetty-7.x.y-SNAPSHOT
2013-08-29 16:53:28.891:INFO::Started SelectChannelConnector@0.0.0.0:8084
1. When I run the test the simulator starts and opens the safari browser, and after approx. 20 seconds I get this error:

07:50:821 WARNING DefaultCommandListener.stderr  2013-08-29 17:07:50.813 instruments[4990:707] Device 'Build Version < Unknown >, < Unknown > ID < Unknown >' is not setup as a development device.  Use Xcode to setup.
java.io.IOException: Bad file descriptor
    at java.io.FileInputStream.readBytes(Native Method)
    at java.io.FileInputStream.read(FileInputStream.java:220)
    at java.lang.UNIXProcess$DeferredCloseInputStream.read(UNIXProcess.java:227)
    at sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:264)
    at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:306)
    at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:158)
    at java.io.InputStreamReader.read(InputStreamReader.java:167)
    at java.io.BufferedReader.fill(BufferedReader.java:136)
    at java.io.BufferedReader.readLine(BufferedReader.java:299)
    at java.io.BufferedReader.readLine(BufferedReader.java:362)
    at org.uiautomation.ios.utils.Command$2.run(Command.java:135)
    at java.lang.Thread.run(Thread.java:680)
java.io.IOException: Stream closed
    at java.io.BufferedInputStream.getBufIfOpen(BufferedInputStream.java:145)
    at java.io.BufferedInputStream.read1(BufferedInputStream.java:255)
    at java.io.BufferedInputStream.read(BufferedInputStream.java:317)
    at sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:264)
    at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:306)
    at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:158)
    at java.io.InputStreamReader.read(InputStreamReader.java:167)
    at java.io.BufferedReader.fill(BufferedReader.java:136)
    at java.io.BufferedReader.readLine(BufferedReader.java:299)
    at java.io.BufferedReader.readLine(BufferedReader.java:362)
    at org.uiautomation.ios.utils.Command$2.run(Command.java:135)
    at java.lang.Thread.run(Thread.java:680)

Since the simulator starts I can't figure out why it's telling me this.
Would you be able to assist please?

**Comment by rgustavsson** [comment]
$ xcodebuild -showsdks
OS X SDKs:
    Mac OS X 10.7                   -sdk macosx10.7
    OS X 10.8                       -sdk macosx10.8

iOS SDKs:
    iOS 6.1                         -sdk iphoneos6.1

iOS Simulator SDKs:
    Simulator - iOS 6.1             -sdk iphonesimulator6.1

**Comment by nicegraham** [comment]
can you launch the simulator manually to make sure it will actually start?

also, you may need to enable developer mode and add yourself to the dev group:
sudo /usr/sbin/DevToolsSecurity --enable
sudo dscl . append /Groups/_developer GroupMembership <username>

**Comment by freynaud** [comment]
i suspect you have an iphone / ipad plugged in. Can you unplug it and try again ?

**Comment by rgustavsson** [comment]
Thanks for the rapid feedback.
When I left work yesterday I powered of and this morning I started the system, opened the terminal, started the ios server and triggered a test.. Guess what? ALL GOOD :)

I guess it was due to that I had an iphone plugged in that wasn't registered as a developer device. My intention wasn't to test on it - only wanted to charge it :)

Many thanks for the help guys. :+1:
